### ISSUE TITLE
dart2js considers "dart.library.io" true

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#27548]
When I compile the following code with `dart2js`:

```dart
import 'dart:core'
    if (dart.library.io) 'dart:io';

void main() {}
```

I get the following output:

```
test.dart:1:1:
Error: Library not supported 'dart:io'.
Try removing the dependency or enabling support using the '--categories' option.
import 'dart:core'
    if (dart.library.io) 'dart:io';

test.dart:
Info: Your app imports the unsupported library 'dart:io' via:

  test.dart => dart:io

Use the --categories option to support import of 'dart:io'.


Error: Compilation failed.
```

I wouldn't expect `dart2js` to try to import `dart:io` here, since it doesn't support it. This happens whether or not I pass `--categories=Client`.

**Comment by nex3** [comment]
Tested against 1.19.1 and 51f617c0d31907a660b7fb6385873242c6a3ad38.

**Comment by sigmundch** [comment]
which version of dart2js are you using?

I believe this was fixed by Lasse in bcdf0a1f125816b97590734370a9e6ef99a10881

**Comment by nex3** [comment]
I just re-tested against 51f617c0d31907a660b7fb6385873242c6a3ad38 and it looks like it works. Sorry for the false alarm!
